tests/func/test_ignore.py

```diff
@@ -112,14 +113,17 @@ def test_ignore_on_branch(tmp_dir, scm, dvc):
     tmp_dir.scm_gen(DvcIgnore.DVCIGNORE_FILE, "foo", commit="add ignore")
 
     scm.checkout("master")
-    assert _files_set(".", dvc.dvcignore) == {"./foo", "./bar"}
+    assert _files_set(".", dvc.tree) == {"./foo", "./bar"}
 
-    dvc.tree = scm.get_tree("branch")
-    assert _files_set(".", dvc.dvcignore) == {"./bar"}
+    tree = CleanTree(scm.get_tree("branch"))
+    files_on_branch = {
+        f for r, ds, fs in tree.walk(tree.tree_root) for f in fs
+    }
+    assert "foo" not in files_on_branch
```

Comment:
`not in` is a very non reliable test, please use `==` instead. Why can't you just:
```python
assert _files_set(".", tree) == {"./bar"}
```

Reply:
One thing here: I changed this assert, but it seems that GitTree always return abs root, which is not how `os.walk` behaves, I think that should be changed, @efiop, @Suor would you agree with me?